### ISSUE TITLE
Microsoft-Rewards not working and left no message...

### ISSUE THREAD

**Issue by Sanjogahara** [charlesbel/Microsoft-Rewards-Farmer#151]
![image](https://user-images.githubusercontent.com/30425702/216835396-7cf45dff-0783-4468-8e51-75ab23848759.png)
I would be really greatful if someone could offer some useful solutions...

**Comment by AlexyGalaxy** [comment]
I have uploaded an updated version of this scrip to my github. It allows for more quizes to be completed as bing rewards keeps updating. I also added timers and counters and im working on adding a logger and error logger.

If you would like to check it out my github link is: https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

**Comment by Sanjogahara** [comment]
> I have uploaded an updated version of this scrip to my github. It allows for more quizes to be completed as bing rewards keeps updating. I also added timers and counters and im working on adding a logger and error logger.
> 
> If you would like to check it out my github link is: https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

Thank you for uploading the updated version. But it seems not to work..Here's the error message:
![image](https://user-images.githubusercontent.com/30425702/217435621-bca4f24e-3e10-4c03-b41c-3237ed124795.png)
By the way, i checked line 779, finding that when using XPATH method it couldn't locate target "//[@id="navs"]/div/div/div/div/div[4]/a'". Maybe the BING web page has changed or it may change with different regions?

**Comment by AlexyGalaxy** [comment]
In my code line 779 does not have waitUntilVisible. Make sure you are using the an up to date version of the script.

However, try changing waitUntilVisible to the code below and let me know what happens. I am located in the US. Where are you located?

```
waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/div/button/span/i', 20)
```

Also disable headless mode by commenting out line 46 and 47 with #. Then watch the script open chrome and look to see where it is stuck. (When done testing make sure to delete the # to rentable headless mode.
```
#if headless_mode : #comment out to disable headless mode (makes window visable) 
           #options.add_argument("--headless") #comment out to disable headless mode (makes window visable) 
```
Pause the script by placing the code located below, before the waitUntilVisible and once again after the waitUntilVisible. Then run the script and post the screen shot here. If the browser is taking too long to load, increase the number at the end of waitUntilVisible to 30-50(this is the time waited)

```
print('[TEST] Press any key to continue')
input()
```

The way this line of code works is by grabing the XPATH information. When your script is paused and waiting for your second input to continue, click on the browsers that was opened and running the script, press F12(this will open a page showing the browsers code), then press ctrl+shift+c, then hover over the top banner of the website and right click. Now in that new page that shows the browsers code there is a line highlighted. Right click that line and select Copy>Copy XPATH. Paste that line of code within waitUntilVisible.
```
waitUntilVisible(browser, By.XPATH, 'PASTE_XPATH_CODE_HERE/DELETE_THESE_WORDS_WITHIN_THE_QUOTES', 20)
```

let me know if that helps. If so please send me the line of code you end up using.

**Comment by Sanjogahara** [comment]
QAQ Never success for me in China...
I've change line 1128 to
`waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/div/button/span/i', 20)`
Here's the shot:
<img width="744" alt="微信图片_20230212134139" src="https://user-images.githubusercontent.com/30425702/218295104-944e5dfe-ac6b-44b4-bced-666c49f3674a.png">
After Grabbing the XPATH information, here's the changed code:
`waitUntilVisible(browser, By.XPATH, '///*[@id="navs"]/div/div/div/div', 20)`

**Comment by Sanjogahara** [comment]
BTY, line 1126 in source code--XPATH '//*[@id="navs"]/div/div/div/div/div[4]/a'--is attmpt to located the "payment and billing" in the top banner of website. No idea if it's changed with region??
`waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/a', 20)`

**Comment by AlexyGalaxy** [comment]
Try this:
Go to https://account.microsoft.com/
Hit f12
Hit ctrl+f in the window showing the browser's code
type nav and look for this
![navs](https://user-images.githubusercontent.com/86795638/218326305-8dccb551-b058-4ce7-87e9-f358a5fb99d8.JPG)

hit the > to expand the tree until you see 4 div and the top bar is highlighted like so:
![img1](https://user-images.githubusercontent.com/86795638/218326357-da35c533-1211-4aa7-be88-c55973472dbe.JPG)
This line of code should be the menubar 
now go 4 lines of code down until "rewards" is highlighted on the menubar
rightclick that line of code (it should be the line below)
![image](https://user-images.githubusercontent.com/86795638/218326610-80e0cc00-4b38-4a18-9522-d76a977c4055.png)

click copy>copy xpath 
for me that line of code is 
//*[@id="navs"]/div/div/div/div/div[4]/a
But I think it might change because of your region.

Now in your ms farmer script, change that line of code and any other lines that have

```
waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/a', 20)
```
to
```
waitUntilVisible(browser, By.XPATH, 'YOUR_CODE_GOES_HERE', 20)
```

Please let me know what works.

**Comment by Sanjogahara** [comment]
Sorry..menubar doesn't have rewards botton in chinese website..
I have to find another script..QAQ
Thank you so much for taking time to help me.

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
